src/market/directory.rs

```diff
@@ -136,12 +158,50 @@ fn write_default_directory_config(config_path: &PathBuf) -> std::io::Result<()>
 
     write_default_config(config_path, config_string)
 }
-
-pub fn start_directory_server(directory: Arc<DirectoryServer>) {
+pub fn start_address_writer_thread(directory: Arc<DirectoryServer>) {
     let address_file = directory.data_dir.join("addresses.dat");
 
-    let addresses = Arc::new(RwLock::new(HashSet::new()));
+    #[cfg(feature = "integration-test")]
+    let interval = 600;
+    #[cfg(not(feature = "integration-test"))]
+    let interval = 60;
```

Comment:
A better way to write it. Also the values will be reversed.

```suggestion
    let interval = if cfg!(feature = "integration-test") {
        60 // 1 min for tests
    } else {
        600 // 10 mins for production
    };
```